tests/testthat/testEdgeThresh.R

```diff
@@ -0,0 +1,251 @@
+test_that(".permuteGraph parameter errors are correctly handled") {
+
+    expect_error(.permuteGraph())
+    expect_error(.permuteGraph(graph = myEGG))
+    expect_error(.permuteGraph(Q = 100))
+
+    expect_error(.permuteGraph(graph = NULL,  Q = 100))
+    expect_error(.permuteGraph(graph = myEGG, Q = NULL))
+
+    expect_error(.permuteGraph(graph = TRUE,  Q = 100))
+    expect_error(.permuteGraph(graph = myEGG, Q = TRUE))
+
+    expect_error(.permuteGraph(graph ="myEGG",Q =  100 ))
+    expect_error(.permuteGraph(graph = myEGG, Q = "100"))
+
+    expect_error(.permuteGraph(graph = myEGG, Q =  0))
+    expect_error(.permuteGraph(graph = myEGG, Q = -1))
+
+    expect_error(.permuteGraph(graph = 10,    Q = 100))
+
+}
+# ((Should valid inputs different from rete.EGGprototype be used?))
+# ((The testing below can get quite computationally expensive...))
+test_that("a sane input to .permuteGraph gives an expected output", {
+
+    ### Test for Q = 1, 10, 20, 50, 100, 200
+    myPermutation <- .permuteGraph(myEGG, Q = 1)
+
+    # Check for self-loops and multiple edges
+    expect_true(igraph::is.simple(myPermutation))
+
+    # Check that the size of the largest connected component has not changed
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+
+    # Check that overall size has not changed
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 10)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 20)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 50)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 100)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 200)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+})
+# ((Is it possible to pass permuted graphs from the previous test section
+# to the next one, if they pass? This would minimize a lot of computation))
+test_that("a sane input to edgeThresh gives an expected output", {
+    # Provide small input data, provide small output data.
```

Comment:
I don't see why you are benchmarking your results by varying Q in this test. You really need to explain to me what your concern is here.